src/hint.c

```diff
@@ -308,6 +308,7 @@ void ReadClientInfo(ClientNode *np, char alreadyMapped)
    if(np->minWidth == np->maxWidth && np->minHeight == np->maxHeight) {
       np->state.border &= ~BORDER_RESIZE;
       np->state.border &= ~BORDER_MAX;
+      np->state.border &= ~BORDER_MIN;
```

Comment:
If it's a dialog, it should have _NET_WM_WINDOW_TYPE_DIALOG set on it, which would hide the minimize button.  I don't like making the assumption that a window that can't be resized can't be minimized.  That would, for example, make it so you couldn't minimize the main window of Audacious (in winamp mode), which would be rather annoying.

I agree with the other change that makes it so dialogs don't have a maximize button.